doc/rtd/reference/datasources/nocloud.rst

```diff
@@ -65,19 +78,32 @@ tell cloud-init where on the filesystem to look.
 
 This argument might look like: ::
 
-  ds=nocloud;s=file://path/to/directory/;h=node-42
+  ds=nocloud;s=file://path/to/directory/
+
+Alternatively, this URI may be defined in a configuration in a file
+:file:`/etc/cloud/cloud.cfg.d/*.cfg` like this: ::
+
+  datasource:
+    NoCloud:
+      seedfrom: file://10.42.42.42/cloud-init/configs/
+
 
 Permitted keys
 ==============
 
-The permitted keys are:
+The only required key is:
 
-* ``h`` or ``local-hostname``
-* ``i`` or ``instance-id``
 * ``s`` or ``seedfrom``
 
-A valid ``seedfrom`` value consists of a URI which must contain a
-trailing ``/``.
+A valid ``seedfrom`` value consists of a URI which must contain a trailing
+``/``. The short key ``s`` is only supported by kernel commandline or
```

Comment:
The short keys/aliases s h and i are all intended only for use on the kernel command-line is it worth separating this "The short keys ``s`` , ``h`` and ``i`` are only supported when reading values from kernel commandline or SMBIOS. When using a base configuration ``*.cfg`` files, the long key name is required.

Reply:
it would be awkward to describe the optional keys under the section describing the required key, but I can move this out into a note